vnext/Shared/Networking/WinRTHttpResource.cpp

```diff
@@ -297,7 +297,32 @@ fire_and_forget WinRTHttpResource::PerformSendRequest(HttpRequestMessage &&reque
     auto sendRequestOp = self->m_client.SendRequestAsync(coRequest);
     self->TrackResponse(coReqArgs->RequestId, sendRequestOp);
 
-    co_await lessthrow_await_adapter<ResponseOperation>{sendRequestOp};
+    if (coReqArgs->Timeout > 0) {
+      // See https://devblogs.microsoft.com/oldnewthing/20220415-00/?p=106486
+      auto timedOut = std::make_shared<bool>(false);
+      auto sendRequestTimeout = [](auto timedOut, auto milliseconds) -> ResponseOperation {
+        // Convert milliseconds to "ticks" (10^-7 seconds)
+        co_await winrt::resume_after(winrt::Windows::Foundation::TimeSpan{milliseconds * 10000});
```

Comment:
could use something like `duration_cast` to make the conversion clearer

Reply:
Not sure, because "ticks" (10 ^ -7) don't have an alias in `<ratio>`.
Defining an arbitrary `std::ratio` seems like overkill to me, and not clearer to read.